Add dladm_vlan module

xen0l opened the pull request:
##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - New Module Pull Request

##### COMPONENT NAME
dladm_vlan

##### ANSIBLE VERSION
```
2.3
```


##### SUMMARY
Module to manage VLAN interfaces on illumos/Solaris systems.

xen0l committed:
Add dladm_vlan module

privateip commented on lib/ansible/modules/network/illumos/dladm_vlan.py:
```diff
@@ -0,0 +1,188 @@
+
+    module.exit_json(**result)
+
+from ansible.module_utils.basic import *
```
please use explicit import statements

privateip commented on lib/ansible/modules/network/illumos/dladm_vlan.py:
```diff
@@ -0,0 +1,188 @@
+    module.exit_json(**result)
+
+from ansible.module_utils.basic import *
+main()
```
please wrap in `if` block

privateip reviewed (changes requested):
* missing RETURN doc string
* missing METADATA
* please consider adding unit test cases

xen0l committed:
Add ANSIBLE_METADATA
Document return values
Make imports explicit
Use native YAML syntax in EXAMPLES

xen0l commented:
@privateip I addressed you comments. As for unit tests, they will be shared among all modules to a certain degree (I am also speaking about modules, which are already included with Ansible). For that reason, I would like to add them in the next round if possible.

xen0l commented:
ready_for_review
